### DISCUSSION TITLE
Full OpenAPI Definition for api.weather.gov

### DISCUSSION THREAD

**Discussion by StephenClouse** [weather-gov/api#201] [Show and tell]
API currently offers a service definition in [OpenAPI version 3 format](https://www.openapis.org/) (previously known as Swagger), which can be found at https://api.weather.gov/openapi.yaml or https://api.weather.gov/openapi.json. However, the current version of this definition is auto-generated from the backend code and very limited in its expression.

As part of improving our service, we have been working on a more complete definition which will be part of our next release. As we have not determined our next release date, we would like to make this useful data available in the meantime.

The full OpenAPI definition can be found here: https://weather-gov.github.io/api/assets/openapi.yaml

Please note that this is for our version currently in development and may not exactly match the production service; however, there are no significant changes intended to any existing endpoints in the next release.

If you have recommendations for improving this documentation, please pass them along by opening an issue here.

**Comment by jah1007** [comment]
Look for an implementation sometime in late March or early April.

**Comment by StephenClouse** [comment]
Updated as of 2021-03-23

In particular, this reflects the upstream change in alert ID format specified in [SCN 21-01](https://www.weather.gov/media/notification/pdf2/scn21-01cap_handler_v2aab.pdf). AlertId is now an unrestricted string value.
